framework/src/main/java/org/checkerframework/common/wholeprograminference/SceneToStubWriter.java

```diff
@@ -0,0 +1,439 @@
+     *
+     * <p>The stub parser can't parse an enum that's labeled "class", but {@link AClass} doesn't
+     * specify if a class is an enum. So track which classes are enums.
+     */
+    private Set<String> enumSet;
+
+    /**
+     * Stolen from {@code IndexFileWriter}. Prints a literal; used when printing the arguments of
+     * annotations.
+     */
+    private void printValue(AnnotationFieldType aft, Object o) {
+        if (aft instanceof AnnotationAFT) {
+            printAnnotation((Annotation) o);
+        } else if (aft instanceof ArrayAFT) {
+            ArrayAFT aaft = (ArrayAFT) aft;
+            printWriter.print('{');
+            if (!(o instanceof List)) {
+                printValue(aaft.elementType, o);
+            } else {
+                List<?> l = (List<?>) o;
+                // watch out--could be an empty array of unknown type
+                // (see AnnotationBuilder#addEmptyArrayField)
+                if (aaft.elementType == null) {
+                    if (l.size() != 0) {
+                        throw new AssertionError();
+                    }
+                } else {
+                    boolean first = true;
+                    for (Object o2 : l) {
+                        if (!first) {
+                            printWriter.print(',');
+                        }
+                        printValue(aaft.elementType, o2);
+                        first = false;
+                    }
+                }
+            }
+            printWriter.print('}');
+        } else if (aft instanceof ClassTokenAFT) {
+            printWriter.print(aft.format(o));
+        } else if (aft instanceof BasicAFT && o instanceof String) {
+            printWriter.print(Strings.escape((String) o));
+        } else if (aft instanceof BasicAFT && o instanceof Long) {
+            printWriter.print(o.toString() + "L");
+        } else {
+            printWriter.print(o.toString());
+        }
+    }
+
+    /** Prints an annotation in Java source format. */
+    private void printAnnotation(Annotation a) {
+        printWriter.print("@" + a.def().name.substring(a.def().name.lastIndexOf('.') + 1));
+        if (!a.fieldValues.isEmpty()) {
+            printWriter.print('(');
+            boolean first = true;
+            for (Map.Entry<String, Object> f : a.fieldValues.entrySet()) {
+                if (!first) {
+                    printWriter.print(',');
+                }
+                printWriter.print(f.getKey() + "=");
+                printValue(a.def().fieldTypes.get(f.getKey()), f.getValue());
+                first = false;
+            }
+            printWriter.print(')');
+        }
+    }
+
+    /**
+     * Prints all annotations in {@code annos}, separated by spaces. See {@link #printAnnotation}.
+     */
+    private void printAnnotations(Collection<? extends Annotation> annos) {
+        for (Annotation tla : annos) {
+            if (!tla.def.name.contains("+")) {
```

Comment:
When/Why does the name contain `+`?

Reply:
No idea. The JLS, in my reading, suggests that isn't allowed. I borrowed this code from the Annotation File Utilities; do you think it's safe to remove it?

Reply:
I found the following comment in `ClassAnnotatoinSceneReader#getAnnotationDef`
```
        // This is an internal JDK annotation such as jdk.Profile+Annotation .
```
So, I would leave it, but add a comment mentioning where this code was copied from.